### PR TITLE
Enable filter modal

### PR DESCRIPTION
**Pull request by DanielKim1**
Restores filter modal that enables filtering events by event source

https://github.com/NatVIII/rva.rip/assets/22206686/c9bc67ed-b2e2-47a6-a8a4-043298cf51ce

The UX is atrocious. Up to you whether you want to merge this in or use this pull request as a prototype for a better implementation

I think my editor auto-formatted some files. I recommend viewing the diff in hide whitespace mode (https://github.com/NatVIII/rva.rip/pull/68/files?diff=unified&w=1)

### CODE REVIEW

File: components/App.vue [hunk 1]
```diff
@@ -168,19 +60,43 @@ if (process.client) pageWidth.value = window.innerWidth;
...
+    if (name === eventSource.name) {
+      disabledEventSources.set(name, eventSource)
+    } else {
+      newEventSources.push(eventSource)
+    }
+  })
+
+  calendarOptions.value.eventSources = newEventSources
+}
+
```
Comment (DanielKim1) [inline components/App.vue:94]: The core of this PR. Basically adding/removing event sources from `calendarOptions.value.eventSources` adds/removes events from that source.

File: components/App.vue [hunk 2]
```diff
@@ -168,19 +60,43 @@ if (process.client) pageWidth.value = window.innerWidth;
...
```
Comment (DanielKim1) [inline components/App.vue:94]: The core of this PR. Basically, adding/removing event sources from `calendarOptions.value.eventSources` adds/removes events from that source.

**Comment by NatVIII** [comment]
Wonderful work Danny 🫡 will take a look at this in a bit, just wanted to say I really appreciate you taking a look at this

**Comment by NatVIII** [comment]
This has been resolved with the merging of #101, the code you contributed was used >:D thank you so much for making the effort to make this pull request Danny and thanks for the permission to re-use the code to fit the existing Tag system.

**Pull request closed without merging**